src/spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -388,9 +388,7 @@ def compute_refrac_period_violations(
     nb_violations = {}
     rp_contamination = {}
 
-    for i, unit_id in enumerate(sorting.unit_ids):
-        if unit_id not in unit_ids:
-            continue
+    for i, unit_id in enumerate(unit_ids):
```

Comment:
I think that this is dangerous because vector acces will be wrng no ?

Reply:
We need to make sure that the `i` used in `nb_rp_violations`, which is a numpy array, is synced with the `unit_id`. I think it is. 

I've added another test which takes two almost-identical `sorting_analyzer`s. Their only difference is that their units are in a different order, with different labelings. The test checks that the results are independent of the ordering and labelling. So the new test should catch if anything like this is wrong in any of the quality metrics (it's very easy to mess up!).

Reply:
@chrishalcrow you see [here](https://github.com/SpikeInterface/spikeinterface/blob/11ae9aa6227e01fd19908d510d142651b8da1c1f/src/spikeinterface/qualitymetrics/misc_metrics.py#L379) that the `nb_rp_violations` have len `num_units`, which is `len(sorting_analyzer.unit_ids)`.

So this is worng. Either we update `num_units` as `len(unit_ids)` or we revert this to `enumerate(sorting.unit_ids)`

Reply:
Super. I'm improving the tests so that they will catch this and will then update. While doing so I've found that the `drift` metrics don't handle passed sub-units correctly.